core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -92,7 +92,7 @@ class AdminManager(val config: KafkaConfig,
           throw new InvalidRequestException("Both numPartitions or replicationFactor and replicasAssignments were set. " +
             "Both cannot be used at the same time.")
         }
-        val assignments = if (topic.assignments().isEmpty) {
+        lazy val assignments = if (topic.assignments().isEmpty) {
```

Comment:
One of the annoyances is that if `trace` logging is enabled, we will end up initializing `assignments` on line 107 below. One way to fix this could be to move the trace logging, perhaps to be inside `AdminZkClient#validateTopicCreate`. I wonder though if, rather than relying on the mechanics of a lazy variable, it'd be more straightforward to move the `topicExists` check to be before we generate the assignments. What do you think?

Reply:
Well, considering the fact that this is only a minor improvement, seems it's still okay to lose the benefit temporarily when tracing logging is enabled. Does it  make sense?

Reply:
Turning on trace logging shouldn't change the error codes returned from create topics.